pkg/factsengine/entities/fact_value.go

```diff
@@ -24,6 +24,36 @@ type FactValue interface {
 	AsInterface() interface{}
 }
 
+// NewFactValue constructs a FactValue from a nested interface.
+func NewFactValue(factInterface interface{}) (FactValue, error) {
+	switch value := factInterface.(type) {
+	case []interface{}:
+		newList := []FactValue{}
+		for _, value := range value {
+			newValue, err := NewFactValue(value)
+			if err != nil {
+				return nil, err
+			}
+			newList = append(newList, newValue)
+		}
+		return &FactValueList{Value: newList}, nil
+	case map[string]interface{}:
+		newMap := make(map[string]FactValue)
+		for key, value := range value {
+			newValue, err := NewFactValue(value)
+			if err != nil {
+				return nil, err
+			}
+			newMap[key] = newValue
+		}
+		return &FactValueMap{Value: newMap}, nil
+	case bool, int, int32, int64, uint, uint32, uint64, float32, float64, string:
```

Comment:
i would have still divided the cases but I see how this is tempting to do

Reply:
As commented offline, let's leave it as it is and change it if we need a more customized conversions